app/grandchallenge/components/forms.py

```diff
@@ -106,6 +106,10 @@ class MultipleCIVForm(Form):
         SelectUploadWidget,
     }
 
+    @property
+    def interface_field_names(self):
+        yield from self.base_obj.values_for_interfaces.keys()
```

Comment:
Wouldn't this make more sense as a model property?

Reply:
I get why you'd think that initially. That was my first go-to as well. 

However, I quickly realized that the data filtering is dependent on the form, not on the model. This is showcased by the fact that we are adding custom fields that just HAPPEN to have the same name as the model (i.e. 'order' and 'title'). Would you agree?

Reply:
Also, the interface fields are generated in the form. If we'd ever change the way the fields are named or generated: we should also change the references to the fields there.

Reply:
`interface_field_names` returns the slugs for all existing interfaces for a reader study / archive, so those are fixed and do not depend on the form. I think you move this to `ValuesForInterfacesMixin`. The `non_civ_field_names` should indeed remain on the form, I agree with that.

Reply:
Mhh, going to do a bit heavier rework. WIP.